src/lib/eco-router/api.ts

```diff
@@ -53,6 +53,43 @@ export function sortTradesByExecutionPrice(trades: Trade[]) {
   })
 }
 
+/**
+ * Sort trades by price in descending order. Best trades are first.
+ * @param trades list of trades
+ * @returns sorted list of trades in descending order
+ */
+export function sortTradesByAmount(trades: Trade[]) {
+  return trades.sort((a, b) => {
+    if (a === undefined || a === null) {
+      return 1
+    }
+    if (b === undefined || b === null) {
+      return -1
+    }
```

Comment:
Loose equals operator instead of a strict one against `null` implicitly checks for both `null` and `undefined`
Something like `console.log(null == undefined) should yield `true`

```suggestion
    if (a == null)  return 1
    if (b == null)  return -1
```

But I don't fully understand why are we checking this if `trades` is a `Trade` array class, can we get falsy values as trades?

Reply:
I removed this verifications but some null values where reaching this functions, don't know why. It's something we need to take a closer look. That's a nica and tighter solution (my only concern is readability). Will check what types are reaching this function and why.